espflash/src/targets/esp32p4.rs

```diff
@@ -30,9 +30,19 @@ const PARAMS: Esp32Params = Esp32Params::new(
 pub struct Esp32p4;
 
 impl Esp32p4 {
+    /// Check if the magic value contains the specified value
     pub fn has_magic_value(value: u32) -> bool {
         CHIP_DETECT_MAGIC_VALUES.contains(&value)
     }
+
+    #[cfg(feature = "serialport")]
+    /// Check if the connection is USB OTG
+    pub(crate) fn connection_is_usb_otg(&self, connection: &mut Connection) -> Result<bool, Error> {
```

Comment:
This function feels like it belongs on the `Connection` struct, so we should probably expose the necessary details from the `targets` module which can then be referenced there.

Reply:
Agree, also #795 introduced `is_using_usb_serial_jtag`, maybe this should be `is_using_usb_otg`

Reply:
I moved `connection_is_usb_otg` to `Connection` but I;m not sure about the others since most of them are not implemented for all chips